File: m/rls/rlb/stsub.c
/* file -  stsub - test substring */
#include "m:\rid\rider.h"
#include "m:\rid\stdef.h"
/* code -  st_sub - test substring */
int st_sub(
char *mod ,
char *str )
{ char *src ;
  char *dst ;
  for(;;)  {
    while ( *mod != *str) {
      if( *str++ == 0)return 0; }
    src = mod, dst = str;
    for(;;)  {
      if( *src == 0)return 1;
    if( *src++ != *dst++)break; }
    ++str;
  } 
} 

File: m/rls/stsub.r
file	stsub - test substring
include rid:rider
include rid:stdef

code	st_sub - test substring

  func	st_sub
	mod : * char			; model substring to find
	str : * char			; string to search
	()  : int			; fine => model is substring of string
  is	src : * char			; for substring match
	dst : * char			;
					;
      repeat				; big loop
	while *mod ne *str		; find the first matching character
	.. fail if *str++ eq		; exhausted string, not found
	src = mod, dst = str		; found first
	repeat				; compare remainder
	   fine if *src eq		; done, found
	until *src++ ne *dst++		; got another
	++str				; again, just past remainder
      forever				;
  end
